### PR TITLE
[libcpplocate] New port

### PR DESCRIPTION
**Pull request by ekilmer**
New port. Also opened an issue on the project's GitHub https://github.com/cginternals/cpplocate/issues/47

- #### What does your PR fix?  
  N/A

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  ~Fails compilation on UWP. Waiting for response from maintainers to see if we want to investigate.~ ~Fixed with respective patch.~ Patch was reverted due to myself lacking knowledge about UWP. UWP is now marked as unsupported

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Comment by JackBoosY** [comment]
Ready for review now?

**Comment by ekilmer** [comment]
> Ready for review now?

@JackBoosY Yes! Thank you.

File: ports/libcpplocate/fix-install-paths.patch [hunk 1]
```diff
@@ -0,0 +1,78 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 670964e..f19361e 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -123,35 +123,18 @@ if("${CMAKE_INSTALL_PREFIX}" STREQUAL "/usr" OR "${CMAKE_INSTALL_PREFIX}" STREQU
+ endif()
+ 
+ # Installation paths
+-if(UNIX AND SYSTEM_DIR_INSTALL)
```
Comment (JackBoosY) [inline ports/libcpplocate/fix-install-paths.patch:9]: I'd like to only modify this line to `if (1)` to reduce the patch size.

File: ports/libcpplocate/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  SHA512 3ab7074a30b089574f12e44842e4894788e5a19ad2a496c5e8307c479b378171974087ff3050e6dffe6e4fc2c2873310cefb191cb1a41bd6c0a83e32ba46ddcc
+  HEAD_REF master
+  PATCHES
+    fix-install-paths.patch
+)
+
+vcpkg_cmake_configure(
+  SOURCE_PATH ${SOURCE_PATH}
+  OPTIONS
+    -DOPTION_BUILD_TESTS=OFF
```
Comment (JackBoosY) [inline ports/libcpplocate/portfile.cmake:14]:
```suggestion
        -DOPTION_BUILD_TESTS=OFF
```

File: ports/libcpplocate/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  REPO cginternals/cpplocate
+  REF 23aeb69c23848ff52456179374b9dfc1ede357bf # 2.3.0 in CMakeLists.txt "META_VERSION_*"
+  SHA512 3ab7074a30b089574f12e44842e4894788e5a19ad2a496c5e8307c479b378171974087ff3050e6dffe6e4fc2c2873310cefb191cb1a41bd6c0a83e32ba46ddcc
+  HEAD_REF master
+  PATCHES
+    fix-install-paths.patch
+)
+
+vcpkg_cmake_configure(
+  SOURCE_PATH ${SOURCE_PATH}
```
Comment (JackBoosY) [inline ports/libcpplocate/portfile.cmake:12]:
```suggestion
  SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/libcpplocate/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  OPTIONS
+    -DOPTION_BUILD_TESTS=OFF
+)
+
+vcpkg_cmake_install()
```
Comment (JackBoosY) [inline ports/libcpplocate/portfile.cmake:17]:
```suggestion
vcpkg_cmake_install()
vcpkg_copy_pdbs()
```

File: ports/libcpplocate/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/share/cpplocate/cmake")
+
+# Patched liblocate config file needs moving to the correct directory
+file(RENAME
+  "${CURRENT_PACKAGES_DIR}/share/cpplocate/liblocate-config.cmake"
+  "${CURRENT_PACKAGES_DIR}/share/liblocate/liblocate-config.cmake"
```
Comment (JackBoosY) [inline ports/libcpplocate/portfile.cmake:37]: It seems incorrect, `vcpkg_cmake_config_fixup` should copy these config files. Can you please double confirm that?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by ekilmer** [comment]
I was just double-checking some things and now I'm not sure what should be the correct version listing...

The most recent tag `v2.2.0` is here https://github.com/cginternals/cpplocate/commit/01fa9d592e4b84d16d33009f4f977e5e60eab496 but the `META_VERSION_PATCH` is listed as `1`, making the full version `2.2.1`, which doesn't seem correct. I would have expected the `v2.2.0` tag to point to the parent of the referenced commit to mark it as the last commit with that version number, right?

On the other hand, their [published Ubuntu packages](https://launchpad.net/~cginternals/+archive/ubuntu/ppa/+build/23047479) have the latest release as `2.3.0` (like this vcpkg PR) uses the commit that I am using for vcpkg `2.3.0` (https://github.com/cginternals/cpplocate/commit/23aeb69c23848ff52456179374b9dfc1ede357bf), but the [branch named `v2.3`](https://github.com/cginternals/cpplocate/tree/v2.3) is a different commit.

@scheibel (who looks to be the maintainer of cpplocate) could you help clear this up please.

**Comment by ekilmer** [comment]
> Thanks for pursuing an integration into vcpkg!

You're welcome!

> I suggest you use the [v2.3.0](https://github.com/cginternals/cpplocate/releases/tag/v2.3.0) release that I created just know.

@scheibel thank you for creating that tag and explaining the versioning for cpplocate. I appreciate your time and response(s) 😄 

------------------

@JackBoosY I have pushed new commits for vcpkg to use the latest tag (v.2.3.0), so this is now ready for another review. Thank you!

I'm not sure what is making CI fail. Should I do anything to fix that?

**Comment by JackBoosY** [comment]
Looks all passed, our pipeline test has some bugs.

File: ports/libcpplocate/portfile.cmake (hunk 5 above)
Comment (JackBoosY) [reply ports/libcpplocate/portfile.cmake:38]: I know why.
`config.cmake` and `export.cmake` are installed in different folders. That's why `vcpkg_cmake_config_fixup` doesn't do that.

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks @ekilmer for the PR and thanks @scheibel for making such an awesome library :)

**Pull request merged**